### ISSUE TITLE
Add support for backing properties on activated types with Session via a [SessionData] attribute

### ISSUE THREAD

**Issue by DamianEdwards** [aspnet/Mvc#6602]
Equivalent to #6525 but using Session as the backing store instead of `ViewData`

**Comment by hishamco** [comment]
Why we didn't use the `TempData` backed by session instead?!!

**Comment by DamianEdwards** [comment]
Because Session requires extra configuration and is semantically different to TempData. We changed the default TempData provider to cookies because it just works out of the box.

**Comment by hishamco** [comment]
You're right, but i'm asking what is a difference if I used `TempData` backed by session? if it's already configured

**Comment by DamianEdwards** [comment]
TempData has extra behavior, that once the value is read it's deleted. The fact it used to use Session as the backing store by default was an implementation detail.

**Comment by hishamco** [comment]
Got it .. thanks

**Comment by pranavkm** [comment]
@rynowak \ @mkArtakMSFT, @DamianEdwards showed interest in this feature + [CookieData](https://github.com/aspnet/Mvc/issues/6603) if there's time to do this for preview2. These should be relatively low cost once we get https://github.com/aspnet/Mvc/pull/7385 in

**Comment by mkArtakMSFT** [comment]
Actually, I'm going to move this to the Backlog as this ask has been there for a while and there wasn't much community interest in this.
